test_helpers.py

```diff
@@ -163,38 +156,31 @@ def test_part_get_values_for_inverts(self):
         ''' Test part of get_values_for_inverter() function, which is in dbus_service 
         but heavily uses functions in helpers.py. '''
 
-        get_power = get_nested(meter_data_null, self.custpower)
-        power_value = try_get_value(get_power, float, self.custpower_default)
-        if isinstance(power_value, float) or isinstance(power_value, int):
-            power = float(power_value * float(self.custpower_factor))
-        else:
-            power = power_value
-
-        get_pv_yield = get_nested(meter_data_null, self.custtotal)
-        pvyield_value = try_get_value(get_pv_yield, float, self.custtotal_default)
-        if isinstance(pvyield_value, float) or isinstance(pvyield_value, int):
-            pvyield = float(pvyield_value * float(self.custtotal_factor))
-        else:
-            pvyield = pvyield_value
-
-        get_voltage = get_nested(meter_data_null, self.custvoltage)
-        voltage_value = try_get_value(get_voltage, float, self.custvoltage_default)
-        if isinstance(voltage_value, float) or isinstance(voltage_value, int):
-            voltage = float(voltage_value)
-        else:
-            voltage = voltage_value
-
-        # get_dc_voltage = get_nested(meter_data, self.custdcvoltage)
-        # dc_voltage_value = try_get_value(get_dc_voltage, float, self.custdcvoltage_default)
-        # dc_voltage = float(dc_voltage_value)
-        # print()
-
-        get_current = get_nested(meter_data_null, self.custcurrent)
-        current_value = try_get_value(get_current, float, self.custcurrent_default)
-        if isinstance(current_value, float) or isinstance(current_value, int):
-            current = float(current_value)
-        else:
-            current = current_value
+        power = dbus_service.DbusService.get_processed_meter_value(
+            meter_data_null,
+            self.custpower,
+            self.custpower_default,
+            self.custpower_factor
+        )
+
+        pvyield = dbus_service.DbusService.get_processed_meter_value(
+            meter_data_null,
+            self.custtotal,
+            self.custtotal_default,
+            self.custtotal_factor
+        )
+
+        voltage = dbus_service.DbusService.get_processed_meter_value(
+            meter_data_null,
+            self.custvoltage,
+            self.custpower_default,
+        )
+
+        current = dbus_service.DbusService.get_processed_meter_value(
+            meter_data_null,
+            self.custcurrent,
+            self.custpower_default,
+        )
 
```

Comment:
I've lost overview over the tests, atm. Is the new cust/default function tested sufficiently. (talking about integration-like test, not unit test)

Reply:
?

Reply:
The problem is, that we don't have any integration tests for a generic device where this PR solves a certain problem. Ahoy and OpenDTU won't send any unknown values to test it properly. Furthermore, we need to find a better way to do integration test without implementing Testdata / methods inside the code under test.